pkg/operator/controller/controller_ca_configmap.go

```diff
@@ -67,7 +70,7 @@ func (r *reconciler) ensureRouterCAConfigMap(secret *corev1.Secret, ingresses []
 
 // desiredRouterCAConfigMap returns the desired router CA configmap.
 func desiredRouterCAConfigMap(secret *corev1.Secret, ingresses []ingressv1alpha1.ClusterIngress) (*corev1.ConfigMap, error) {
-	if !shouldPublishRouterCA(ingresses) {
+	if secret == nil || !shouldPublishRouterCA(ingresses) {
```

Comment:
I don't think you need to check `secret == nil` both here and in `ensureRouterCAConfigMap`.  The pattern I have been following is that `ensureFoo` is responsible for checking prerequisite values and will return immediately if any are missing, so `desiredFoo` can safely assume the prerequisites are present.  Does that seem reasonable?

Reply:
I don't think we want to enforce restrictions that this method is only called by ensureXYZ method. Each method handling its own basic checks will be fool proof during refactors and makes it reusable like  `desiredDNSRecords`.